Sources/PagerStripViewController.swift

```diff
@@ -0,0 +1,356 @@
+        }
+    }
+
+    open func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        if containerView == scrollView {
+            lastPageNumber = pageFor(contentOffset: scrollView.contentOffset.x)
+        }
+    }
+
+    open func scrollViewDidEndScrollingAnimation(_ scrollView: UIScrollView) {
+        if containerView == scrollView {
+            pagerTabStripChildViewControllersForScrolling = nil
+            (navigationController?.view ?? view).isUserInteractionEnabled = true
+            updateContent()
+        }
+    }
+
+    // MARK: - Orientation
+
+    open override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        isViewRotating = true
+        pageBeforeRotate = currentIndex
+        coordinator.animate(alongsideTransition: nil) { [weak self] _ in
+            guard let me = self else { return }
+            me.isViewRotating = false
+            me.currentIndex = me.pageBeforeRotate
+            me.preCurrentIndex = me.currentIndex
+            me.updateIfNeeded()
+        }
+    }
+
+    // MARK: Private
+
+    private func progressiveIndicatorData(_ virtualPage: Int) -> (Int, Int, CGFloat) {
+        let count = viewControllers.count
+        var fromIndex = currentIndex
+        var toIndex = currentIndex
+        let direction = swipeDirection
+
+        if direction == .left {
+            if virtualPage > count - 1 {
+                fromIndex = count - 1
+                toIndex = count
+            } else {
+                if self.scrollPercentage >= 0.5 {
+                    fromIndex = max(toIndex - 1, 0)
+                } else {
+                    toIndex = fromIndex + 1
+                }
+            }
+        } else if direction == .right {
+            if virtualPage < 0 {
+                fromIndex = 0
+                toIndex = -1
+            } else {
+                if self.scrollPercentage > 0.5 {
+                    fromIndex = min(toIndex + 1, count - 1)
+                } else {
+                    toIndex = fromIndex - 1
+                }
+            }
+        }
+
+        return (fromIndex, toIndex, self.scrollPercentage)
+    }
+
+    private func reloadViewControllers() {
+        guard let dataSource = datasource else {
+            fatalError("dataSource must not be nil")
+        }
+        viewControllers = dataSource.viewControllers(for: self)
+
+        guard !viewControllers.isEmpty else {
+            fatalError("viewControllers(for:) should provide at least one child view controller")
+        }
+        viewControllers.forEach { if !($0 is IndicatorInfoProvider) { fatalError("Every view controller provided by PagerTabStripDataSource's viewControllers(for:) method must conform to IndicatorInfoProvider") }}
+
+    }
+
+    private var pagerTabStripChildViewControllersForScrolling: [UIViewController]?
```

Comment:
I think properties, even private, should be declared at the top. Is there a particular reason why you prefer them at the bottom?